src/main/java/com/sap/oss/phosphor/fosstars/data/release/info/ReleaseInfoLoader.java

```diff
@@ -0,0 +1,92 @@
+package com.sap.oss.phosphor.fosstars.data.release.info;
+
+import com.sap.oss.phosphor.fosstars.data.AbstractReleaseInfoLoader;
+import com.sap.oss.phosphor.fosstars.data.DataProvider;
+import com.sap.oss.phosphor.fosstars.data.github.GitHubDataFetcher;
+import com.sap.oss.phosphor.fosstars.data.github.ReleasesFromGitHub;
+import com.sap.oss.phosphor.fosstars.model.Artifact;
+import com.sap.oss.phosphor.fosstars.model.Value;
+import com.sap.oss.phosphor.fosstars.model.ValueSet;
+import com.sap.oss.phosphor.fosstars.model.release.info.MavenArtifact;
+import com.sap.oss.phosphor.fosstars.model.release.info.NpmArtifact;
+import com.sap.oss.phosphor.fosstars.model.subject.oss.GitHubProject;
+import com.sap.oss.phosphor.fosstars.model.value.ArtifactVersions;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * This data provider tries to fill out the
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature. It is based on one of the following data providers:
+ * <ul>
+ * <li>{@link ReleaseInfoFromMaven}</li>
+ * <li>{@link ReleaseInfoFromNpm}</li>
+ * </ul>
+ * The data provider uses {@link ReleasesFromGitHub} as a fallback if
+ * {@link com.sap.oss.phosphor.fosstars.model.feature.oss.OssFeatures#RELEASED_ARTIFACT_VERSIONS}
+ * feature could not be filled up from one of the above-mentioned data providers.
+ */
+public class ReleaseInfoLoader extends AbstractReleaseInfoLoader<Artifact> {
+
+  /**
+   * A {@link ReleasesFromGitHub} data provider.
+   */
+  private final DataProvider gitHubReleaseInfoProvider;
+
+  /**
+   * Initializes a data provider.
+   *
+   * @param fetcher An interface to GitHub.
+   */
+  public ReleaseInfoLoader(GitHubDataFetcher fetcher) {
+    Objects.requireNonNull(fetcher, "Oh no! You gave me a null instead of a GitHub fetcher!");
+    gitHubReleaseInfoProvider = new ReleasesFromGitHub(fetcher);
+  }
+
+  @Override
+  public DataProvider<Artifact> update(Artifact artifact, ValueSet values) throws IOException {
+    DataProvider provider = providerFor(artifact);
+    provider.update(artifact, values);
+
+    for (Value<?> value : values) {
+      if (value.get() instanceof ArtifactVersions && !((ArtifactVersions) value.get()).empty()) {
+        break;
+      }
+
+      if (artifact.gitHubProject().isPresent()) {
+        doUpdate(artifact.gitHubProject().get(), values);
+        break;
+      }
```

Comment:
If I understand correctly, the data provider just needs to call one of the underlying data providers. The choice depends on the artifact. Should the logic look like the following?

```suggestion
      if (artifact instanceof MavenArtifact) {
        releaseInfoFromMaven.update((MavenArtifact) artifact, newValues);
      } else if (artifact instanceof NpmArtifact) {
        releaseInfoFromNpm.update((NpmArtifact) artifact, newValues);
      } else if (artifact.gitHubProject().isPresent()) {
        gitHubReleaseInfoProvider.update(project, newValues);
      } else {
        // what should it do here? Just quit? Or, report a error? Set RELEASED_ARTIFACT_VERSIONS to unknown?
      }
      // if values already have RELEASED_ARTIFACT_VERSIONS, should it be replaced? or merged? 
```

Reply:
This is the scenario:
- Check if `NPM` or `MAVEN` related release information could be collected.
- Only if no information is collected. Then check in `GitHubProjectInfo`.
- If still no information is collected then set the Feature value to unknown.

The release information has to be merged depending on what information is already available and not available. Still unsure of how to achieve this at this point.

Reply:
I am looking for something like this 

```
      if (artifact instanceof MavenArtifact) {
        releaseInfoFromMaven.update((MavenArtifact) artifact, newValues);
      } else if (artifact instanceof NpmArtifact) {
        releaseInfoFromNpm.update((NpmArtifact) artifact, newValues);
      }
      
      if (values(RELEASED_ARTIFACT_VERSIONS).elements.isEmpty() && artifact.gitHubProject().isPresent()) {
        gitHubReleaseInfoProvider.update(project, newValues);
      } 
      
      // if still nothing do nothing.

```

Reply:
You can easily merge these if-blocks and call `gitHubReleaseInfoProvider` in another if-else-block.

Reply:
> The release information has to be merged depending on what information is already available and not available. Still unsure of how to achieve this at this point.

If values already has `RELEASED_ARTIFACT_VERSIONS`, you can just merge `ArtifactVersions`. Do you see any problem here?